.github/workflows/tests.yaml

```diff
@@ -0,0 +1,86 @@
+name: Test
+
+permissions:
+  contents: read
+
+on: [push, pull_request]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ startsWith(github.ref, 'refs/pull/') }}
+
+jobs:
+  test:
+    name: "Tests (${{ matrix.compiler }}, C++${{ matrix.standard}}, ${{ matrix.mode }}, ${{ matrix.enables }})"
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        # only the compilers supported by setup-cpp
+        compiler: [clang++-18, gcc-13]
+        standard: [20, 23]
+        mode: [dev, debug, release]
+        include:
+          # only build this combination with dpdk enabled, so we don't double
+          # the size of the test matrix, and can at least test the build with
+          # dpdk enabled.
+          - compiler: clang++-18
+            standard: 23
+            mode: release
+            cooks: --cook dpdk
+            enables: --enable-dpdk
+          # only build this combination with C++20 moduels enabled, so we don't double
+          # the size of the test matrix, and can at least test the build with
+          # C++20 modules enabled.
+          - compiler: clang++-18
+            standard: 23
+            mode: debug
+            enables: --enable-cxx-modules
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: "${{ contains(matrix.cooks, 'dpdk') }}"
+
+      - name: Install build dependencies
+        run: |
+          sudo ./install-dependencies.sh
+
+      # instal the latest cmake and ninja when building C++ modules
```

Comment:
typo: install

Why do we need the latest cmake and ninja only when building modules? What do we use if we're not building modules?

Reply:
because the cmake shipped by the runner image (ubuntu:jammy) does not support C++ modules. the proper C++module is officially supported in cmake 3.28 and ninja 1.11. jammy comes with cmake 3.22, see https://packages.ubuntu.com/jammy/cmake.

for CMake related document, see also https://cmake.org/cmake/help/latest/manual/cmake-cxxmodules.7.html.